### PR TITLE
fix: remove notification from state on open

### LINKED ISSUE #785: Clicking a notification marks it as read but doesn't remove it from the app
### 🔍 Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### 📝 Description

Using the `main` branch, when you click on a notification it automatically gets marked as read, but Gitify doesn't remove the notification until the 60s setInterval gets triggered again

Should we remove the notification on click?

### 🪜 Steps To Reproduce

1. Build & run the app from the `main` branch
2. Click on a notification
3. Wait for the next notifications refresh -> the notification disappears

### ℹ System Info

```Markdown
`main` branch
```

### 📸 Screenshots

_No response_

### PR DESCRIPTION
**Pull request by adufr**
## Related issues

- Fixes #785 

## Context

Remove notification from state when clicking on it.

### CODE REVIEW

**Review by setchy — DISMISSED** [review]
lgtm, just needs some more coverage

**Comment by adufr** [comment]
I'm trying to test add a test that would like like this but it doesn't work:
```typescript
  describe('removeNotificationFromState', () => {
    it('should remove a notification from state', async () => {
      const { result } = renderHook(() => useNotifications(false));
      result.current.notifications = mockedSingleAccountNotifications;
      result.current.removeNotificationFromState(mockedSingleAccountNotifications[0].notifications[0].id, mockedSingleAccountNotifications[0].hostname);

      expect(result.current.notifications[0].notifications.length).toBe(0);
    });
  });
``` 

I'm not very familiar with React so I'd appreciate some help 😬

**Comment by adufr** [comment]
Coverage has now increased 🚀

**Review by setchy — APPROVED** [review]
Thanks @adufr

File: src/components/NotificationRow.tsx [hunk 1]
```diff
@@ -33,6 +34,9 @@ export const NotificationRow: React.FC<IProps> = ({
 
     if (settings.markAsDoneOnOpen) {
       markNotificationDone(notification.id, hostname);
+    } else {
+      // no need to mark as read, github does it by default when opening it
```
Comment (bmulholland) [inline src/components/NotificationRow.tsx:38]: Thanks for the explanation!

**Pull request merged**